docs: upgrade guide improvements (#2166)
FusionAuth/fusionauth-site: site/docs/v1/tech/installation-guide/_theme-upgrade.adoc

In the version release notes, any new templates and macros are documented. If there are changes, you'll want to take a closer look at the themes post upgrade.

As part of your upgrade testing, open the administrative user interface and navigate to [breadcrumb]#Customizations > Themes#.

If any themes are missing templates, they will show as "Upgrade Required". Port the new theme files over to your custom theme, modify them as needed, and save the theme.

In some cases, existing templates are modified. If you have customized your theme, you'll need to compare the new template to your existing version's base theme and port over any changes to your customized theme. The easiest way to do this is to use a diff tool to compare the two sets of files. Here is a suggested process to follow before you upgrade:

1. Download the default theme from your existing version of FusionAuth.
2. Download the default theme from the new version of FusionAuth.
3. Use a diff tool to compare the two sets of files. 
4. Apply any differences to your customized theme.

> You can use the link:https://github.com/FusionAuth/fusionauth-theme-helper[Theme Helper] to help  with this process.

Reviewer: ```suggestion
You can use the https://github.com/FusionAuth/fusionauth-theme-helper[Theme Helper] to help  with this process.
```

==== Using the Theme Helper to Upgrade Themes

> Clone the link:https://github.com/FusionAuth/fusionauth-theme-helper[Theme Helper] repo and follow the install instructions in the `README.md` file. 

Reviewer: ```suggestion
Clone the https://github.com/FusionAuth/fusionauth-theme-helper[Theme Helper] repo and follow the install instructions in the `README.md` file. 
```

Download the base themes from your existing version of FusionAuth and the new version of FusionAuth to compare.

To get the existing version's theme files, create a `.env` file from the `.env.sample` file. Create an link:/docs/v1/tech/apis/authentication#managing-api-keys[API key] with read permissions for Themes in the FusionAuth administrative user interface. Add the API key to the `.env` file. The FusionAuth link:/docs/v1/tech/reference/limitations#default-configuration[default theme has the same ID `75a068fd-e94b-451a-9aeb-3ddb9a3b5987`] across all instances. Use this value to update the `THEME_ID` key in the `.env` file.

In the `.env` file, set the `FUSIONAUTH_URL` to the URL of your existing FusionAuth instance. Finally, update the `TMP_DIR` to a directory on your local machine where you want the existing theme files to be downloaded, such as `current-theme`.

Now you can run the `download.sh` script to download the existing theme files to the `TMP_DIR` directory.

> Once the download is complete, you'll need to get the base theme files from the new version of FusionAuth. The easiest way to do this before upgrading your production version is to install the new version of FusionAuth on your local machine or a VM using Docker. Instructions for installing FusionAuth using Docker can be found in the link:/docs/v1/tech/installation-guide/docker#docker-compose[FusionAuth Docker Installation Guide].

Reviewer: ```suggestion
Once the download is complete, you'll need to get the base theme files from the new version of FusionAuth. The easiest way to do this is to install the new version of FusionAuth on your local machine or a VM using Docker. Instructions for installing FusionAuth using Docker can be found in the link:/docs/v1/tech/installation-guide/docker#docker-compose[FusionAuth Docker Installation Guide].
```

> Once you've got the new version of FusionAuth running, you can update the Theme Helper `.env` file in the Theme Helper repo to point to the new version of FusionAuth. You'll need to update the `FUSIONAUTH_URL` to `\http://localhost:9011`. 

Reviewer: ```suggestion
Once you've got the new version of FusionAuth running, you can update the Theme Helper `.env` file in the Theme Helper repo to point to the new version of FusionAuth. If running locally, update the `FUSIONAUTH_URL` to `\http://localhost:9011`. 
```
